docs/HARDWARE-MODEL.md

```diff
@@ -0,0 +1,154 @@
+
+## Generating initial model file
+
+When EVE-OS is booted on a device one can run a script to get a partial and initial model file. That script is invoked as:
+
+```shell
+# eve exec debug spec.sh
+```
+
+Note that the script does not take photos of the front and back, nor find the product URL, hence the output must be edited. The less obvious but more critical manual work includes:
+
+- The script looks for controllers on the PCI bus but has no visibility to which controllers are connected to which external physical ports. This is a an issue in particular for USB ports where one controller can be connected to multiple physical ports, but there can also be multiple USB controllers perhaps with some of them having no external physical ports. On a device with multiple USB controllers manual testing is needed to determine which controller is connected to which external USB ports.
+- A cellular modem (if available) is typically connected via a USB controller, but it might not be visible (not plugged in or software not yet initializing it) when you run the script. ```lsusb``` and ```lspci``` can help finding those.
+- There can be additional functions on the PCI bus which the script does not look for, and such functions can limit the ability to assign other adapters/functions to application instances. More about that below.
+
+### Completing the model file
+
+Starting from the generated model file and add or adjust:
+
+- Add the product URL
+- Add photos of the front and back with sufficient detail to show any printed labels next to the physical ports. If there are no such unique labels, add annotations on the photo with unique labels.
+- Edit the phylabel fields in the model file to match the above labels.
+- Optionally, if the intended application pre-determined and e.g., eth1 will be connected to the shopfloor network, you can set the logicallabel field to "shopfloor". That makes it more clear in the UI the intended use of that port.
+- If the model file shows multiple USB controllers, then plug in e.g., a USB stick in each physical USB port and use ```lsusb``` and ```lspci``` to tell which controller to which port is connected.
+- If the device has multiple physical USB ports, please add a entry for each one of them in the json file (on the correct USB controller if there are multiple)
+- If you device has a cellular modem verify that there is one or two wwan interfaces in the generated file, and if not add one. If it is connected via a USB controller (```lsusb``` will tell you that) it needs to be in the same assignment group as the USB controller.
+- Check whether there are additional functions on the PCI controllers which might prevent assignment to an application instance. See below.
+
+#### Check other functions
+
+All of the iommu groups can be determined as shown in this example:
```

Comment:
You should make a note that this only works on KVM

Reply:
Updated.